grpc-gateway/pb/hubConfiguration_test.go

```diff
@@ -0,0 +1,25 @@
+package pb
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDeviceOAuthClient_Clone(t *testing.T) {
+	a := DeviceOAuthClient{
+		ProviderName: "test",
+		Scopes:       []string{"test"},
+	}
+	b := a.Clone()
+	require.Equal(t, &a, b)
```

Comment:
The intent of the Clone functions seems to be to do a deep copy, right? I'd extend the tests to check that the variables do not share pointers, thus modifying `a` has no effect on `b`.

Reply:
strings can share the pointers

Reply:
only structs needs to copied

Reply:
added check for arrays

Reply:
> strings can share the pointers

Yes, but if you change the string in a it doesn't effect the string in b. Apparently, strings and arrays will have a Clone method in golang v1.20.

Reply:
👍